components/ProjectForm/index.js

```diff
@@ -0,0 +1,127 @@
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm();
+
+  const onSubmit = (data) => {
+    onAddProject(data);
+    console.log(projects);
+    reset();
+    router.push("/");
+  };
+
+  return (
+    <>
+      <StyledForm onSubmit={handleSubmit(onSubmit)}>
+        <h2>Create new project</h2>
+        <label>
+          Project title
+          <input
+            {...register("title", { required: "Title is required" })}
+            placeholder="Title"
+          />
+          <StyledErrorMessage>{errors.title?.message}</StyledErrorMessage>
+        </label>
+
+        <label>
+          Image
+          <input {...register("image")} placeholder="Select Image" />
+        </label>
+
+        <label>
+          Description
+          <textarea
+            {...register("description", {
+              required: "Description is required",
+            })}
+            placeholder="Description"
+          />
+          <StyledErrorMessage>{errors.description?.message}</StyledErrorMessage>
+        </label>
+        <div>
+          <label>
+            Duration
+            <select {...register("duration")} defaultValue={"select..."}>
+              <option value="select...">select...</option>
+              <option value="short">short</option>
+              <option value="medium">medium</option>
+              <option value="long">long</option>
+            </select>
+          </label>
+          &nbsp;
+          <label>
+            Difficulty
+            <select {...register("difficulty")} defaultValue={"select..."}>
+              <option value="select...">select...</option>
+              <option value="easy">easy</option>
+              <option value="medium">medium</option>
+              <option value="hard">hard</option>
+            </select>
+          </label>
+        </div>
+        <label>
+          Materials
+          <input
+            {...register("material.0.amount", {
+              required: "Amount is required",
+            })}
+            type="number"
+            placeholder="Number"
+            min="1"
+          />
+          <StyledErrorMessage>
+            {errors.material_amount?.message}
+          </StyledErrorMessage>
+          <input
+            {...register("material.0.material", {
+              required: "Material is required",
+            })}
+            placeholder="Material"
+          />
+          <StyledErrorMessage>{errors.material?.message}</StyledErrorMessage>
+        </label>
+
+        <label>
+          Instructions
+          <textarea
+            {...register("instruction.0.steps", {
+              required: "Instructions are required",
+            })}
+            placeholder="Instructions"
+          />
+          <StyledErrorMessage>
+            {errors.instructions?.message}
+          </StyledErrorMessage>
+        </label>
+        <div>
+          <Link href="/">
+            <button>Cancel</button>
+          </Link>
```

Comment:
We do not combine `link` and `button` as they have different purposes. You can style the `link` to look like a button. :) [for an article about difference between `button` and `link` see CSS Tricks](https://css-tricks.com/a-complete-guide-to-links-and-buttons/)